static/js/server_events_dispatch.js

```diff
@@ -223,11 +223,15 @@ export function dispatch_normal_event(event) {
                             // TODO: Add waiting_period_threshold logic here.
                             page_params.can_create_streams =
                                 page_params.is_admin ||
+                                (page_params.is_moderator &&
+                                    page_params.realm_create_stream_policy !== 2) ||
                                 page_params.realm_create_stream_policy === 1;
```

Comment:
These bits of logic are getting complex enough that I kinda want to move these calculations to functions in `settings_data.js` and stop using the `page_params` parameter at all.  (That will make it reasonable to write tests for them, etc.).  And then I think we can just have the places we need this state just call those functions rather than accessing `page_params`, and delete the `server_events_dispatch.js` logic here.

(Can all be a prep commit).

Reply:
@timabbott I understand that by this change we would remove direct usage of `page_params.can_create_streams` from other places in code also, like where we actually send the information to template for showing the create stream button.
But due to this we would want to add the logic for checking whether the user is new member or not here also.

Reply:
Yeah -- I think we only would call this function in two places, so it'd be a pretty simple refactor:
```
$ git grep page_params[.]can_create_streams
frontend_tests/node_tests/hotkey.js:    page_params.can_create_streams = true;
frontend_tests/node_tests/hotkey.js:    page_params.can_create_streams = false;
static/js/hotkey.js:        if (event_name === "n_key" && overlays.streams_open() && page_params.can_create_streams) {
static/js/server_events_dispatch.js:                            page_params.can_create_streams =
static/js/subs.js:            can_create_streams: page_params.can_create_streams,
```

But you're right -- we'd need to implement the `waiting_period_threshold` logic correctly -- i.e. address the TODO where the logic we have now is wrong.